### PR TITLE
AG-147 - Spring module initial commit

### PR DESCRIPTION
**Pull request by barreiro**

### CODE REVIEW

File: agroal-springframework/src/main/java/io/agroal/springframework/AgroalDataSourceConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2020 Red Hat, Inc. and individual contributors as indicated by the @author tags.
+// You may not use this file except in compliance with the Apache License, Version 2.0.
+
+package io.agroal.springframework;
```
Comment (geoand) [inline agroal-springframework/src/main/java/io/agroal/springframework/AgroalDataSourceConfiguration.java:4]: As this is spring boot specific, maybe the package name should reflect that?

File: agroal-springframework/pom.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            <artifactId>spring-boot-autoconfigure</artifactId>
+            <version>${version.org.springframework.boot}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-jpa</artifactId>
+            <version>${version.org.springframework.boot}</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (geoand) [inline agroal-springframework/pom.xml:41]: Are you sure this dependency is needed?

File: agroal-springframework/src/main/java/io/agroal/springframework/AgroalDataSourceConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.transaction.jta.JtaTransactionManager;
+
+import javax.sql.DataSource;
+
+/**
+ * @author <a href="lbarreiro@redhat.com">Luis Barreiro</a>
+ */
+@Configuration
```
Comment (geoand) [inline agroal-springframework/src/main/java/io/agroal/springframework/AgroalDataSourceConfiguration.java:21]: You might want to use:

```
@Configuration(proxyBeanMethods = false)
``` 

Which is the norm these days

**Review by geoand — APPROVED** [review]
LGTM, but I would like @gytis to take a look as well as he is the SB+Narayana SME

File: agroal-springframework/pom.xml (hunk 2 above)
Comment (gytis) [reply agroal-springframework/pom.xml:41]: Seems that it should be enough to have `spring-tx` no?

Comment (barreiro) [reply agroal-springframework/pom.xml]: Changed dependency to `spring-tx`. Updated `narayana-spring-boot-starter` to `2.4.0` and added test case to check the transaction integration is engaged. Thanks @gytis for the quick fix!

File: agroal-springframework/src/main/java/io/agroal/springframework/AgroalDataSourceConfiguration.java (hunk 1 above)
Comment (barreiro) [reply agroal-springframework/src/main/java/io/agroal/springframework/AgroalDataSourceConfiguration.java]: Changed module name to `agroal-spring-boot-starter` and package name to `io.agroal.springframework.boot`

File: agroal-springframework/src/main/java/io/agroal/springframework/AgroalDataSourceConfiguration.java (hunk 3 above)
Comment (barreiro) [reply agroal-springframework/src/main/java/io/agroal/springframework/AgroalDataSourceConfiguration.java]: Good suggestion. Applied.

**Pull request merged**